src/core/ext/filters/client_channel/lb_policy/outlier_detection/outlier_detection.cc

```diff
@@ -176,27 +176,34 @@ class OutlierDetectionLb : public LoadBalancingPolicy {
 
       void Eject() {
         ejected_ = true;
-        if (last_seen_state_.has_value() &&
-            *last_seen_state_ != GRPC_CHANNEL_TRANSIENT_FAILURE) {
-          watcher_->OnConnectivityStateChange(GRPC_CHANNEL_TRANSIENT_FAILURE);
+        if (last_seen_state_.has_value()) {
```

Comment:
is this change for nor checking the last_seen_state value intentional?

Reply:
Yes.  The outlier detection policy indicates an ejected subchannel by reporting TRANSIENT_FAILURE for that subchannel instead of the real connectivity state.  This means that it's possible to have "duplicate" TRANSIENT_FAILURE notifications if we eject a subchannel whose real state was already TRANSIENT_FAILURE or if we uneject a subchannel whose real state is TRANSIENT_FAILURE.

Previously, we were avoiding sending these duplicate notifications by checking whether the real state was already TRANSIENT_FAILURE.  However, now that we're sending the status along with the connectivity state, the "duplicate" notifications aren't actually duplicates anymore, because even though the connectivity state isn't changing, the status is.  And we don't want to avoid updating the status, because that will lead to incorrect RPC failure status messages -- e.g., we might report "subchannel ejected by outlier detection" when the real problem is actually a connectivity issue, or vice versa.  To avoid that, I've changed the code here to send the notifications on ejection and unejection even if the connectivity state is not changing.